CI: codecov failures for test_petab_test_suite.yml

[Workflow](https://github.com/AMICI-dev/AMICI/blob/bac5d025966e7949f088d6eac7a9244389b55b5f/.github/workflows/test_petab_test_suite.yml) [fails to find](https://github.com/AMICI-dev/AMICI/runs/6212039344?check_suite_focus=true ) coverage files:
```
[2022-04-28T13:53:24.924Z] ['error'] None of the following appear to exist as files: ./coverage.xml
[2022-04-28T13:53:24.925Z] ['error'] There was an error running the uploader: Error while cleaning paths. No paths matched existing files!
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2/dist/codecov' failed with exit code 255
```
Unclear what changed. That file still gets created by pytest-cov according to the logs.

@thomasrockhu-codecov I did, full output here: 
```
2022-05-05T16:31:08.7527608Z ##[group]Run codecov/codecov-action@v3.1.0
2022-05-05T16:31:08.7527961Z with:
2022-05-05T16:31:08.7528489Z   token: ***
2022-05-05T16:31:08.7528697Z   file: coverage.xml
2022-05-05T16:31:08.7528888Z   flags: petab
2022-05-05T16:31:08.7529088Z   fail_ci_if_error: true
2022-05-05T16:31:08.7529291Z   verbose: true
2022-05-05T16:31:08.7529476Z env:
2022-05-05T16:31:08.7529660Z   ENABLE_GCOV_COVERAGE: true
2022-05-05T16:31:08.7529948Z   pythonLocation: /opt/hostedtoolcache/Python/3.8.12/x64
2022-05-05T16:31:08.7530339Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.12/x64/lib
2022-05-05T16:31:08.7530682Z   BNGPATH: /home/runner/work/AMICI/AMICI/ThirdParty/BioNetGen-2.7.0
2022-05-05T16:31:08.7530939Z ##[endgroup]
2022-05-05T16:31:08.9118359Z ==> linux OS detected
2022-05-05T16:31:09.4098615Z https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
2022-05-05T16:31:09.5684431Z ==> SHASUM file signed by key id 806bb28aed779869
2022-05-05T16:31:09.7562214Z ==> Uploader SHASUM verified (cae21e490c24b05c3802c7001a1540a7f83d46cb4f1eadf14e76627094c65cb8  codecov)
2022-05-05T16:31:09.7564445Z ==> Running version latest
2022-05-05T16:31:09.8507320Z ==> Running version v0.2.1
2022-05-05T16:31:09.8529470Z [command]/home/runner/work/_actions/codecov/codecov-action/v3.1.0/dist/codecov -n  -Q github-action-3.1.0 -Z -f coverage.xml -F petab -C 33915f4bdc950b20e812632ca4f5cb373f14cabe -v
2022-05-05T16:31:10.0576788Z [2022-05-05T16:31:10.056Z] ['verbose'] Start of uploader: 1651768270056...
2022-05-05T16:31:10.0583222Z [2022-05-05T16:31:10.058Z] ['info'] 
2022-05-05T16:31:10.0583662Z      _____          _
2022-05-05T16:31:10.0587532Z     / ____|        | |
2022-05-05T16:31:10.0590242Z    | |     ___   __| | ___  ___ _____   __
2022-05-05T16:31:10.0590572Z    | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
2022-05-05T16:31:10.0590800Z    | |___| (_) | (_| |  __/ (_| (_) \ V /
2022-05-05T16:31:10.0591036Z     \_____\___/ \__,_|\___|\___\___/ \_/
2022-05-05T16:31:10.0591189Z 
2022-05-05T16:31:10.0591329Z   Codecov report uploader 0.2.0
2022-05-05T16:31:10.0647182Z [2022-05-05T16:31:10.064Z] ['info'] => Project root located at: /home/runner/work/AMICI/AMICI
2022-05-05T16:31:10.0682839Z [2022-05-05T16:31:10.068Z] ['info'] ->  Token found by environment variables
2022-05-05T16:31:10.0692351Z [2022-05-05T16:31:10.069Z] ['verbose'] Start of network processing...
2022-05-05T16:31:10.0694161Z [2022-05-05T16:31:10.069Z] ['verbose'] Searching for files in /home/runner/work/AMICI/AMICI
2022-05-05T16:31:10.1624797Z [2022-05-05T16:31:10.162Z] ['info'] Running coverage xml...
2022-05-05T16:31:10.8582848Z [2022-05-05T16:31:10.857Z] ['info'] Searching for coverage files...
2022-05-05T16:31:11.3417244Z [2022-05-05T16:31:11.341Z] ['verbose'] Preparing to clean the following coverage paths: coverage.xml
2022-05-05T16:31:11.3421889Z [2022-05-05T16:31:11.341Z] ['error'] None of the following appear to exist as files: coverage.xml
2022-05-05T16:31:11.3422635Z [2022-05-05T16:31:11.341Z] ['error'] There was an error running the uploader: Error while cleaning paths. No paths matched existing files!
2022-05-05T16:31:11.3426904Z [2022-05-05T16:31:11.342Z] ['verbose'] The error stack is: Error: Error while cleaning paths. No paths matched existing files!
2022-05-05T16:31:11.3427424Z     at cleanCoverageFilePaths (/snapshot/repo/dist/src/helpers/files.js)
2022-05-05T16:31:11.3427835Z     at main (/snapshot/repo/dist/src/index.js)
2022-05-05T16:31:11.3428253Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-05-05T16:31:11.3428784Z [2022-05-05T16:31:11.342Z] ['verbose'] End of uploader: 1286 milliseconds
2022-05-05T16:31:11.3519934Z ##[error]Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3.1.0/dist/codecov' failed with exit code 255
```

Thanks, @thomasrockhu-codecov. It looks like `0.2.2` is used, but I still see get the same [output](https://github.com/AMICI-dev/AMICI/runs/6313922780?check_suite_focus=true):

```
2022-05-05T21:36:31.6857216Z ##[group]Run codecov/codecov-action@v3.1.0
2022-05-05T21:36:31.6857506Z with:
2022-05-05T21:36:31.6858356Z   token: ***
2022-05-05T21:36:31.6858609Z   file: coverage.xml
2022-05-05T21:36:31.6858868Z   flags: petab
2022-05-05T21:36:31.6859104Z   fail_ci_if_error: true
2022-05-05T21:36:31.6859360Z   verbose: true
2022-05-05T21:36:31.6859592Z env:
2022-05-05T21:36:31.6859821Z   ENABLE_GCOV_COVERAGE: true
2022-05-05T21:36:31.6860172Z   pythonLocation: /opt/hostedtoolcache/Python/3.8.12/x64
2022-05-05T21:36:31.6860574Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.12/x64/lib
2022-05-05T21:36:31.6860990Z   BNGPATH: /home/runner/work/AMICI/AMICI/ThirdParty/BioNetGen-2.7.0
2022-05-05T21:36:31.6861337Z ##[endgroup]
2022-05-05T21:36:31.8674109Z ==> linux OS detected
2022-05-05T21:36:32.2717035Z https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
2022-05-05T21:36:32.3528973Z ==> SHASUM file signed by key id 806bb28aed779869
2022-05-05T21:36:32.5741595Z ==> Uploader SHASUM verified (cae21e490c24b05c3802c7001a1540a7f83d46cb4f1eadf14e76627094c65cb8  codecov)
2022-05-05T21:36:32.5743465Z ==> Running version latest
2022-05-05T21:36:32.6284883Z ==> Running version v0.2.2
2022-05-05T21:36:32.6314226Z [command]/home/runner/work/_actions/codecov/codecov-action/v3.1.0/dist/codecov -n  -Q github-action-3.1.0 -Z -f coverage.xml -F petab -C 3f7aff0644c4baaf7c5bb02a8b4ec5eeb48b66d2 -v
2022-05-05T21:36:32.8915198Z [2022-05-05T21:36:32.889Z] ['verbose'] Start of uploader: 1651786592889...
2022-05-05T21:36:32.8921084Z [2022-05-05T21:36:32.891Z] ['info'] 
2022-05-05T21:36:32.8921604Z      _____          _
2022-05-05T21:36:32.8922814Z     / ____|        | |
2022-05-05T21:36:32.8923241Z    | |     ___   __| | ___  ___ _____   __
2022-05-05T21:36:32.8923640Z    | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
2022-05-05T21:36:32.8924063Z    | |___| (_) | (_| |  __/ (_| (_) \ V /
2022-05-05T21:36:32.8924441Z     \_____\___/ \__,_|\___|\___\___/ \_/
2022-05-05T21:36:32.8924701Z 
2022-05-05T21:36:32.8924928Z   Codecov report uploader 0.2.0
2022-05-05T21:36:32.8986416Z [2022-05-05T21:36:32.898Z] ['info'] => Project root located at: /home/runner/work/AMICI/AMICI
2022-05-05T21:36:32.9028552Z [2022-05-05T21:36:32.902Z] ['info'] ->  Token found by environment variables
2022-05-05T21:36:32.9039225Z [2022-05-05T21:36:32.903Z] ['verbose'] Start of network processing...
2022-05-05T21:36:32.9040068Z [2022-05-05T21:36:32.903Z] ['verbose'] Searching for files in /home/runner/work/AMICI/AMICI
2022-05-05T21:36:33.0176558Z [2022-05-05T21:36:33.017Z] ['info'] Running coverage xml...
2022-05-05T21:36:33.8317441Z [2022-05-05T21:36:33.830Z] ['info'] Searching for coverage files...
2022-05-05T21:36:34.3777141Z [2022-05-05T21:36:34.377Z] ['verbose'] Preparing to clean the following coverage paths: coverage.xml
2022-05-05T21:36:34.3782114Z [2022-05-05T21:36:34.377Z] ['error'] None of the following appear to exist as files: coverage.xml
2022-05-05T21:36:34.3783051Z [2022-05-05T21:36:34.377Z] ['error'] There was an error running the uploader: Error while cleaning paths. No paths matched existing files!
2022-05-05T21:36:34.3786282Z [2022-05-05T21:36:34.378Z] ['verbose'] The error stack is: Error: Error while cleaning paths. No paths matched existing files!
2022-05-05T21:36:34.3786919Z     at cleanCoverageFilePaths (/snapshot/repo/dist/src/helpers/files.js)
2022-05-05T21:36:34.3787420Z     at main (/snapshot/repo/dist/src/index.js)
2022-05-05T21:36:34.3787933Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-05-05T21:36:34.3788578Z [2022-05-05T21:36:34.378Z] ['verbose'] End of uploader: 1489 milliseconds
2022-05-05T21:36:34.3903890Z ##[error]Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3.1.0/dist/codecov' failed with exit code 255
```

--- a/.github/workflows/test_petab_test_suite.yml
+++ b/.github/workflows/test_petab_test_suite.yml
@@ -71,7 +71,8 @@ jobs:
       - name: Run PEtab-related unit tests
         run: |
           source ./build/venv/bin/activate \
-            && pytest --cov-report=xml --cov=./ python/tests/test_*petab*.py
+            && pytest --cov-report=xml:coverage.xml \
+              --cov=./ python/tests/test_*petab*.py
 
       # run test models
       - name: Run PEtab test suite
@@ -79,12 +80,15 @@ jobs:
         run: |
           source ./build/venv/bin/activate \
           && AMICI_PARALLEL_COMPILE=2 pytest -v \
-            --cov-report=xml --cov-append --cov=amici tests/petab_test_suite/
+            --cov-report=xml:coverage.xml \
+            --cov-append \
+            --cov=amici \
+            tests/petab_test_suite/
 
       - name: Codecov
         uses: codecov/codecov-action@v3.1.0
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
-          file: ./coverage.xml
+          file: coverage.xml
           flags: petab
           fail_ci_if_error: true

--- a/.github/workflows/test_petab_test_suite.yml
+++ b/.github/workflows/test_petab_test_suite.yml
@@ -82,7 +82,7 @@ jobs:
             --cov-report=xml --cov-append --cov=amici tests/petab_test_suite/
 
       - name: Codecov
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v3.1.0
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           file: ./coverage.xml

--- a/.github/workflows/test_petab_test_suite.yml
+++ b/.github/workflows/test_petab_test_suite.yml
@@ -92,5 +92,3 @@ jobs:
           file: coverage.xml
           flags: petab
           fail_ci_if_error: true
-          verbose: true
-          version: v0.2.2

--- a/.github/workflows/test_petab_test_suite.yml
+++ b/.github/workflows/test_petab_test_suite.yml
@@ -92,3 +92,4 @@ jobs:
           file: coverage.xml
           flags: petab
           fail_ci_if_error: true
+          verbose: true

--- a/.github/workflows/test_petab_test_suite.yml
+++ b/.github/workflows/test_petab_test_suite.yml
@@ -93,3 +93,4 @@ jobs:
           flags: petab
           fail_ci_if_error: true
           verbose: true
+          version: v0.2.2

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -130,6 +130,7 @@ jobs:
         flags: cpp
         fail_ci_if_error: true
         verbose: true
+        version: v0.2.2
 
     - name: Run sonar-scanner
       env:

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -129,8 +129,6 @@ jobs:
         file: coverage.info
         flags: cpp
         fail_ci_if_error: true
-        verbose: true
-        version: v0.2.2
 
     - name: Run sonar-scanner
       env:

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -104,7 +104,7 @@ jobs:
         scripts/runNotebook.sh documentation/GettingStarted.ipynb
 
     - name: Codecov Python
-      uses: codecov/codecov-action@v2
+      uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
         file: ./build/coverage_py.xml
@@ -122,7 +122,7 @@ jobs:
         && lcov -a coverage_cpp.info -a coverage_py.info -o coverage.info
 
     - name: Codecov CPP
-      uses: codecov/codecov-action@v2
+      uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
         file: ./coverage.info

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -107,7 +107,7 @@ jobs:
       uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
-        file: ./build/coverage_py.xml
+        file: build/coverage_py.xml
         flags: python
         fail_ci_if_error: true
 
@@ -125,7 +125,7 @@ jobs:
       uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
-        file: ./coverage.info
+        file: coverage.info
         flags: cpp
         fail_ci_if_error: true
 

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -110,6 +110,7 @@ jobs:
         file: build/coverage_py.xml
         flags: python
         fail_ci_if_error: true
+        verbose: true
 
     - name: lcov
       run: |
@@ -128,6 +129,7 @@ jobs:
         file: coverage.info
         flags: cpp
         fail_ci_if_error: true
+        verbose: true
 
     - name: Run sonar-scanner
       env:

--- a/.github/workflows/test_sbml_semantic_test_suite.yml
+++ b/.github/workflows/test_sbml_semantic_test_suite.yml
@@ -53,7 +53,7 @@ jobs:
         path: tests/amici-semantic-results
 
     - name: Codecov SBMLSuite
-      uses: codecov/codecov-action@v2
+      uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
         file: ./coverage_SBMLSuite.xml

--- a/.github/workflows/test_sbml_semantic_test_suite.yml
+++ b/.github/workflows/test_sbml_semantic_test_suite.yml
@@ -56,6 +56,6 @@ jobs:
       uses: codecov/codecov-action@v3.1.0
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
-        file: ./coverage_SBMLSuite.xml
+        file: coverage_SBMLSuite.xml
         flags: sbmlsuite
         fail_ci_if_error: true

